File: solutions/fsharp/rna-transcription/RnaTranscription.fs
module RnaTranscription

let toRna (dna: string): string =
    let transcription = function
        | 'G' -> 'C'
        | 'C' -> 'G'
        | 'T' -> 'A'
        | 'A' -> 'U'
        | other -> other

    dna |> String.map transcription

File: solutions/scala/rna-transcription/src/main/scala/RnaTranscription.scala
object RnaTranscription {
  private val DnaRnaMap = Map(
    'C' -> 'G', 'G' -> 'C', 'T' -> 'A', 'A' -> 'U'
  )

  def toRna(dna: String): Some[String] =
    Some(dna.map(DnaRnaMap))
}